CMakeLists.txt

```diff
@@ -1,82 +1,132 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.2)
 project(usb_cam)
 
-## Find catkin macros and libraries
-## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
-## is used, also find other catkin packages
-find_package(catkin REQUIRED COMPONENTS cv_bridge image_transport roscpp std_msgs std_srvs sensor_msgs camera_info_manager)
-
-# Find OpenCV
-find_package(OpenCV REQUIRED)
-
-## pkg-config libraries
+message(STATUS "Searching required modules")
 find_package(PkgConfig REQUIRED)
+# FFMPEG
 pkg_check_modules(avcodec libavcodec REQUIRED)
 pkg_check_modules(avutil libavutil REQUIRED)
 pkg_check_modules(swscale libswscale REQUIRED)
+pkg_check_modules(video4linux libv4l2 REQUIRED)
+# OPENCV
+find_package(OpenCV REQUIRED)
+# Reporting
+if(avcodec_FOUND)
+    message(STATUS "Found libavcodec: ${avcodec_VERSION}")
+endif()
+if(avutil_FOUND)
+    message(STATUS "Found libavutil: ${avutil_VERSION}")
+endif()
+if(swscale_FOUND)
+    message(STATUS "Found libswscale: ${swscale_VERSION}")
+endif()
+if(OpenCV_FOUND)
+    message(STATUS "Found OpenCV: ${OpenCV_VERSION}")
+endif()
+if(video4linux_FOUND)
+    message(STATUS "Found Video4Linux kernel library: ${video4linux_VERSION}")
+endif()
 
-###################################################
-## Declare things to be passed to other projects ##
-###################################################
-
-## LIBRARIES: libraries you create in this project that dependent projects also need
-## CATKIN_DEPENDS: catkin_packages dependent projects also need
-## DEPENDS: system dependencies of this project that dependent projects also need
+# Catkin
+find_package(catkin REQUIRED COMPONENTS
+    cv_bridge
+    image_transport
+    roscpp
+    std_msgs
+    std_srvs
+    sensor_msgs
+    camera_info_manager
+)
 catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
+    INCLUDE_DIRS include
+    LIBRARIES usb_cam
 )
 
-###########
-## Build ##
-###########
+# Build scenario
+option(SUPPRESS_BIND_GLOBAL_PLACEHOLDERS
+    "Suppresses Boost >= 1.68 deprecation warning message for global placeholders in boost::bind instantiation"
+    ON
+) # Workaround for deprecation message from instantiation of Boost::bind
+if(SUPPRESS_BIND_GLOBAL_PLACEHOLDERS)
+    add_compile_definitions(BOOST_BIND_GLOBAL_PLACEHOLDERS)
```

Comment:
`add_compile_definitions` is from `cmake` `3.12`.
https://cmake.org/cmake/help/latest/command/add_compile_definitions.html

this cmakelists file requires only `3.2`, and melodic's `cmake` is `3.10.2`.
so these lines should be like this.

```
if("$ENV{ROS_DISTRO}" STRGREATER "melodic")
  if(SUPPRESS_BIND_GLOBAL_PLACEHOLDERS)
      add_compile_definitions(BOOST_BIND_GLOBAL_PLACEHOLDERS)
  endif()
endif()
```

Reply:
@knorth55 thank you for the note, implemented